runelite-client/src/main/java/net/runelite/client/plugins/timers/TimersPlugin.java

```diff
@@ -618,6 +638,113 @@ else if (event.getMessage().startsWith(KILLED_TELEBLOCK_OPPONENT_TEXT))
 				}
 			}
 		}
+
+		if (config.showTzhaarTimers())
+		{
+			String message = Text.removeTags(event.getMessage());
+			Matcher matcher = TZHAAR_COMPLETE_MESSAGE.matcher(message);
+
+			if (message.contains(TZHAAR_DEFEATED_MESSAGE) || matcher.matches())
+			{
+				removeTzhaarTimer();
+				resetTzhaarConfig();
+				resetTzhaarVars();
+				return;
+			}
+
+			Instant now = Instant.now();
+			matcher = TZHAAR_PAUSED_MESSAGE.matcher(message);
+			if (matcher.matches())
+			{
+				tzhaarLastTime = now;
+				createTzhaarTimer(tzhaarStartTime, now);
+				return;
+			}
+
+			matcher = TZHAAR_WAVE_MESSAGE.matcher(message);
+			if (!matcher.matches())
+			{
+				return;
+			}
+
+			if (!tzhaarStarted)
+			{
+				int wave = Integer.parseInt(matcher.group(1));
+				if (wave != 1)
+				{
+					return;
+				}
+
+				tzhaarStarted = true;
+				tzhaarStartTime = now;
+			}
+
+			else if (tzhaarLastTime != null)
+			{
+				tzhaarStartTime = tzhaarStartTime.plus(Duration.between(tzhaarStartTime, now)).minus(Duration.between(tzhaarStartTime, tzhaarLastTime));
+				tzhaarLastTime = null;
+			}
+
+			createTzhaarTimer(tzhaarStartTime, tzhaarLastTime);
+		}
+
+	}
+
+	private void updateInfoBoxState()
+	{
+		if (tzhaarTimer == null)
+		{
+			return;
+		}
+
+		if (!checkInFightCaves() && !checkInInferno())
+		{
+			removeTzhaarTimer();
+			resetTzhaarConfig();
+			resetTzhaarVars();
+		}
+	}
+
+	private boolean checkInFightCaves()
+	{
+		return client.getMapRegions() != null && ArrayUtils.contains(client.getMapRegions(), 9551);
+	}
+
+	private boolean checkInInferno()
+	{
+		return client.getMapRegions() != null && ArrayUtils.contains(client.getMapRegions(), 9043);
+	}
+
+	private void resetTzhaarVars()
+	{
+		tzhaarStartTime = null;
+		tzhaarLastTime = null;
+		tzhaarStarted = false;
+	}
+
+	private void removeTzhaarTimer()
+	{
+		infoBoxManager.removeInfoBox(tzhaarTimer);
+		tzhaarTimer = null;
+	}
+
+	private void createTzhaarTimer(Instant startTime, Instant lastTime)
+	{
+		if (tzhaarTimer != null)
+		{
+			infoBoxManager.removeInfoBox(tzhaarTimer);
+		}
+
+		if (checkInFightCaves())
+		{
+			tzhaarTimer = new ElapsedTimer(itemManager.getImage(FIRE_CAPE), this, startTime, lastTime);
```

Comment:
Would the Jad and Zuk pets be better images for these timers?

Reply:
I don't think so, because slayer task uses the bosses as icons already, so if you were on task you'd have 2x jad or 2x zuk infoboxes